src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_Utilities.cs

```diff
@@ -156,72 +161,39 @@ static bool Contains(LspDocumentSymbol parent, LspDocumentSymbol child)
             // Converts a Document Symbol Range to a SnapshotSpan within the text snapshot used for the LSP request.
             SnapshotSpan GetSymbolRangeSpan(Range symbolRange)
             {
-                var originalStartPosition = originalSnapshot.GetLineFromLineNumber(symbolRange.Start.Line).Start.Position + symbolRange.Start.Character;
-                var originalEndPosition = originalSnapshot.GetLineFromLineNumber(symbolRange.End.Line).Start.Position + symbolRange.End.Character;
+                var originalStartPosition = textSnapshot.GetLineFromLineNumber(symbolRange.Start.Line).Start.Position + symbolRange.Start.Character;
+                var originalEndPosition = textSnapshot.GetLineFromLineNumber(symbolRange.End.Line).Start.Position + symbolRange.End.Character;
 
-                return new SnapshotSpan(originalSnapshot, Span.FromBounds(originalStartPosition, originalEndPosition));
+                return new SnapshotSpan(textSnapshot, Span.FromBounds(originalStartPosition, originalEndPosition));
             }
         }
-
         /// <summary>
-        /// Sorts and returns an immutable array of DocumentSymbolData based on a SortOption.
+        /// Converts an immutable array of DocumentSymbolData to an immutable array of <see cref="DocumentSymbolDataViewModel"/>.
         /// </summary>
-        public static ImmutableArray<DocumentSymbolData> SortDocumentSymbolData(
-            ImmutableArray<DocumentSymbolData> documentSymbolData,
+        public static ImmutableArray<DocumentSymbolDataViewModel> GetDocumentSymbolItemViewModels(
             SortOption sortOption,
-            CancellationToken cancellationToken)
+            ImmutableArray<DocumentSymbolData> documentSymbolData)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-
-            // Log which sort option was used
-            Logger.Log(sortOption switch
+            using var _ = ArrayBuilder<DocumentSymbolDataViewModel>.GetInstance(documentSymbolData.Length, out var documentSymbolItems);
+            foreach (var documentSymbol in documentSymbolData)
             {
-                SortOption.Name => FunctionId.DocumentOutline_SortByName,
-                SortOption.Location => FunctionId.DocumentOutline_SortByOrder,
-                SortOption.Type => FunctionId.DocumentOutline_SortByType,
-                _ => throw new NotImplementedException(),
-            }, logLevel: LogLevel.Information);
+                var children = GetDocumentSymbolItemViewModels(sortOption, documentSymbol.Children);
+                var documentSymbolItem = new DocumentSymbolDataViewModel(documentSymbol, children);
+                documentSymbolItems.Add(documentSymbolItem);
+            }
 
-            return SortDocumentSymbols(documentSymbolData, sortOption, cancellationToken);
+            documentSymbolItems.Sort(DocumentSymbolDataViewModelSorter.GetComparer(sortOption));
```

Comment:
It looked like we're doing the sorting in the WPF collection view now, so do we still need this?

Reply:
good question.  @jmarolf ?